druid/examples/either_prism.rs

```diff
@@ -0,0 +1,126 @@
+use druid::widget::{Checkbox, Flex, Label, Slider};
+use druid::{AppLauncher, Data, Lens, LocalizedString, Widget, WidgetExt, WindowDesc};
+
+use druid::Prism;
+use druid::{
+    BoxConstraints, Env, Event, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx, PaintCtx, UpdateCtx,
+};
+use druid::{LensExt, PrismExt};
+
+#[derive(Clone, Default, Data, Lens)]
+struct AppState {
+    value: f64,
+    panel: SliderOrLabel,
+}
+
+#[derive(Clone, Data, Prism)]
+pub enum SliderOrLabel {
+    Slider {
+        // TODO
+        save_value: bool,
+    },
+    Label(String),
+}
+
+impl From<bool> for SliderOrLabel {
+    fn from(b: bool) -> Self {
+        match b {
+            false => Self::Slider { save_value: false },
+            true => Self::default(),
+        }
+    }
+}
+
+impl From<&SliderOrLabel> for bool {
+    fn from(sol: &SliderOrLabel) -> Self {
+        match &sol {
+            SliderOrLabel::Slider { .. } => false,
+            SliderOrLabel::Label(_) => true,
+        }
+    }
+}
+
+impl Default for SliderOrLabel {
+    fn default() -> Self {
+        Self::Label("Click to reveal slider".into())
+    }
+}
+
+impl Widget<SliderOrLabel> for Checkbox {
```

Comment:
I think there should be some kind of widget wrapper here doing the adaptation of checkbox to the enum.
This pattern of implementing From on boolean then reimplementing Widget seems quite verbose and hard to explain.

Reply:
Yes, I agree. I'm not comfortable with the current example implementation, and I plan to have / believe there are better ways to implement it. I'd say the purpose of this example is to be temporary, and to show how/why it should be improved.

Reply:
I'm not sure if this example is ever getting better, but I copied/adapted an example from `druid-enums`' `login.rs` example, which is more likely to work.